src/cephadm/cephadm

```diff
@@ -971,10 +978,27 @@ def extract_uid_gid(img='', file_path='/var/lib/ceph'):
     (uid, gid) = out.split(' ')
     return (int(uid), int(gid))
 
+def deploy_extra_files(extra, data_dir, uid, gid):
+    # type: (dict[str, list[str]], str, int, int) -> dict[str, str]
+    mounts = {}
+    if extra:
+        # There are extra files to place in the data dir
+        for filename, data in extra.items():
+            host_filename = os.path.basename(filename)
+            host_filename = os.path.join(data_dir, 'extra', host_filename)
+            if isinstance(data, (list, tuple)):
+                data = "\n".join(data)
```

Comment:
```
error: Incompatible types in assignment (expression has type "str", variable has type "List[str]")
```

Maybe deal with the line endings while doing the file write?

```suggestion
        # There are extra files to place in the data dir
        for filename, data in extra.items():
            host_filename = os.path.basename(filename)
            host_filename = os.path.join(data_dir, 'extra', host_filename)
            makedirs(os.path.dirname(host_filename), uid, gid, 0o755)
            with open(host_filename, 'w') as f:
                os.fchown(f.fileno(), uid, gid)
                os.fchmod(f.fileno(), 0o600)
                for line in data:
                    if not line.endswith(os.linesep):
                        line += os.linesep
                    f.write(line)
            mounts[host_filename] = filename
```

Reply:
From what I've read, Python is smart enough to use the correct line endings so we only need '\n'. And I assumed if the file has line endings we want them retained (think things like authorised_hosts etc). If there wasn't suppose to be any line endings, then we'd either have 1 string in the list or just a string (which is what the isinstance(.., (list, tuple)) is catching.

Reply:
Aslo, ok so we're doing tests in string typing, so can't just reuse a variable and change it's type, good to know, I can fix these :)